### PR TITLE
Fixed bug with non integer disk sizes

### PR DESCRIPTION
**Pull request by X4mp**
Signed-off-by: Roland Urbano <rurbano@anexia-it.com>

### Description

To support the possibility to receive non-integer disk sizes an additional field for floating point disk sizes was added to the virtual server. The field is `compute` only and cannot be configured, but is only used for update checks and comparison.

### Acceptance tests
- [x] Have you added an acceptance test for the functionality being added?
- [x] Have you run the acceptance tests on this branch?

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAnxCloudVirtualServer'
...
```

### Release Note
Release note for [CHANGELOG](https://github.com/anexia-it/terraform-provider-anxcloud/blob/master/CHANGELOG.md):

```release-note
* resource/virtual_server - fixed bug with disk sizing
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

File: anxcloud/struct_virtual_server.go [hunk 1]
```diff
@@ -268,16 +271,21 @@ func flattenVirtualServerInfo(in *info.Info) []interface{} {
 	return []interface{}{att}
 }
 
-func flattenVirtualServerDisks(in []vm.Disk) []interface{} {
+func flattenVirtualServerDisks(in []Disk) []interface{} {
 	att := make([]interface{}, len(in))
 
 	for i, d := range in {
 		net := map[string]interface{}{}
```
Comment (X4mp) [inline anxcloud/struct_virtual_server.go:278]: Rename `net` to `disk`.

File: anxcloud/struct_virtual_server_test.go [hunk 2]
```diff
@@ -164,7 +165,7 @@ func TestExpanderVirtualServerInfo(t *testing.T) {
 							"storage_type":   "SSD",
 							"bus_type":       "SCSI",
 							"bus_type_label": "SCSI(0:0) Hard disk 1",
-							"disk_gb":        90,
+							"disk_gb":        90.00,
```
Comment (X4mp) [inline anxcloud/struct_virtual_server_test.go:168]: Decrease test disk size to a smaller amount.

File: anxcloud/resource_virtual_server.go [hunk 3]
```diff
@@ -409,15 +413,15 @@ func resourceVirtualServerUpdate(ctx context.Context, d *schema.ResourceData, m
...
+				addDisks = append(addDisks, *newDisks[i].Disk)
 				continue
 			}
 
 			actualDisk := oldDisks[i]
 			expectedDisk := newDisks[i]
 
-			if actualDisk.Type != expectedDisk.Type || actualDisk.SizeGBs != expectedDisk.SizeGBs {
-				changeDisks = append(changeDisks, expectedDisk)
+			if actualDisk.Type != expectedDisk.Type || actualDisk.ExactDiskSize < float64(expectedDisk.SizeGBs) {
```
Comment (X4mp) [inline anxcloud/resource_virtual_server.go:423]: Add docstring explaining why we compare with the exact disk size.

File: anxcloud/struct_virtual_server.go [hunk 4]
```diff
@@ -268,16 +271,21 @@ func flattenVirtualServerInfo(in *info.Info) []interface{} {
...
 		net["disk_gb"] = d.SizeGBs
 		net["disk_id"] = d.ID
+		net["disk_exact"] = d.ExactDiskSize
 		att[i] = net
 	}
 
 	return att
 }
+
+func roundDiskSize(size float64) int {
```
Comment (X4mp) [inline anxcloud/struct_virtual_server.go:289]: Add tests to be sure how the rounding is done.

**Review by X4mp — COMMENTED** [review]
First review round done with @paepke and @HaveFun83

File: anxcloud/struct_virtual_server_test.go (hunk 2 above)
Comment (X4mp) [reply anxcloud/struct_virtual_server_test.go:168]: those are just unittest. Nothing is being created with these specs.

File: Makefile [hunk 5]
```diff
@@ -2,11 +2,11 @@ export GOPATH?=$(shell go env GOPATH)
 export GOPROXY=https://proxy.golang.org
 export GO111MODULE=on
 
-HOSTNAME=hashicorp.com
+HOSTNAME=hashicorp.local
```
Comment (paepke) [inline Makefile:5]: .local is wrong here

**Pull request merged**